Document/0x06c-Reverse-Engineering-and-Tampering.md

```diff
@@ -609,6 +609,373 @@ Start Safari on the iOS device. Run the above Python script on your connected ho
+
+In the following sections you will be using r2frida to retrieve information straight from the app runtime. First start by opening an r2frida session to the target app (e.g. iGoat-Swift) that should be running on your iPhone (connected per USB). Use the following command:
+
+```bash
+$ r2 frida://usb//iGoat-Swift
+```
+
+> See all options with `r2 frida://?`.
+
+Once in the r2frida session, all commands start with `\`. For example, in radare2 you'd run `i` to display the binary information, but in r2frida you'd use `\i`.
+
+###### Memory Maps and Inspection
+
+You can retrieve the app's memory maps by running `\dm`:
+
+```bash
+[0x00000000]> \dm
+0x0000000100b7c000 - 0x0000000100de0000 r-x /private/var/containers/Bundle/Application/3ADAF47D-A734-49FA-B274-FBCA66589E67/iGoat-Swift.app/iGoat-Swift
+0x0000000100de0000 - 0x0000000100e68000 rw- /private/var/containers/Bundle/Application/3ADAF47D-A734-49FA-B274-FBCA66589E67/iGoat-Swift.app/iGoat-Swift
+0x0000000100e68000 - 0x0000000100e97000 r-- /private/var/containers/Bundle/Application/3ADAF47D-A734-49FA-B274-FBCA66589E67/iGoat-Swift.app/iGoat-Swift
+...
+0x0000000100ea8000 - 0x0000000100eb0000 rw-
+0x0000000100eb0000 - 0x0000000100eb4000 r--
+0x0000000100eb4000 - 0x0000000100eb8000 r-x /usr/lib/TweakInject.dylib
+0x0000000100eb8000 - 0x0000000100ebc000 rw- /usr/lib/TweakInject.dylib
+0x0000000100ebc000 - 0x0000000100ec0000 r-- /usr/lib/TweakInject.dylib
+0x0000000100f60000 - 0x00000001012dc000 r-x /private/var/containers/Bundle/Application/3ADAF47D-A734-49FA-B274-FBCA66589E67/iGoat-Swift.app/Frameworks/Realm.framework/Realm
+```
+
+While you're searching or exploring the app memory, you can always verify where your current offset is located in the memory map. Instead of noting and searching for the memory address in this list you can simply run `\dm.`. You'll find an example in the following section "In-Memory Search".
+
+If you're only interested into the modules (binaries and libraries) that the app has loaded, you can use the command `\il` to list them all:
+
+```shell
+[0x00000000]> \il
+0x0000000100b7c000 iGoat-Swift
+0x0000000100eb4000 TweakInject.dylib
+0x00000001862c0000 SystemConfiguration
+0x00000001847c0000 libc++.1.dylib
+0x0000000185ed9000 Foundation
+0x000000018483c000 libobjc.A.dylib
+0x00000001847be000 libSystem.B.dylib
+0x0000000185b77000 CFNetwork
+0x0000000187d64000 CoreData
+0x00000001854b4000 CoreFoundation
+0x00000001861d3000 Security
+0x000000018ea1d000 UIKit
+0x0000000100f60000 Realm
+```
+
+As you might expect you can correlate the addresses of the libraries with the memory maps: e.g. the main app binary iGoat-Swift is located at `0x0000000100b7c000` and the Realm Framework at `0x0000000100f60000`.
+
+You can also use objection to display the same information.
+
+```shell
+$ objection --gadget OWASP.iGoat-Swift explore
```

Comment:
dont' forget to guide the user on how to use objection; or is that chapter close by?